engine/runtime-parser/src/test/java/org/enso/compiler/core/IrPersistanceTest.java

```diff
@@ -508,4 +509,44 @@ public record RefHolder(Persistance.Reference<UUID> id) {
       this(Persistance.Reference.of(id));
     }
   }
+
+  @Persistable(id = 432877)
+  public static class Looper {
+    public Seq<Looper> seq;
+
+    public Seq<Looper> seq() {
+      return seq;
+    }
+
+    public Looper(Seq<Looper> seq) {
+      this.seq = seq;
+    }
+  }
+
+  @Ignore
+  @Test
+  public void testLoop1() throws Exception {
+    Looper looper = new Looper(null);
+    looper.seq = join(looper, nil());
+
+    Looper out = serde(Looper.class, looper, -1);
+    assertSame(out, out.seq().apply(0));
+  }
+
+  @Ignore
```

Comment:
These tests are tricky. `Seq` is _eager on serialization_ and only **lazy on deserialization**. As such these tests are testing something that is not really working per (current) design. I don't want to change `Seq` to be _lazy on serialization_ as that would expand the "indirect references" table at the end of the file. OK to disable/remove these tests, @radeusgd?

Reply:
Yes it is probably fine.

I think I was adding them because the practical use-case I have in the Types PR relied on Seq and I thought that it could be possible for it to work. But as long as it is possible to rewrite that use-case to still work (which I assume should be possible by breaking the cycle through a `Persistance.Reference` in there), it will be all I need.

Reply:
OK, the `Seq` looping test removed in 35feca4

> relied on `Seq` .. possible by breaking the cycle through a `Persistance.Reference`

Yes, `Persistance.Reference` shall help you do what's needed. I don't want to modify `Seq` to use `Persitance.Reference` behind the scene, as that comes with additional cost and could significantly increases the size of generated `.bindings` caches.

Reply:
> Yes, `Persistance.Reference` shall help you do what's needed. 

Awesome!

> I don't want to modify `Seq` to use `Persitance.Reference` behind the scene, as that comes with additional cost and could significantly increases the size of generated `.bindings` caches.

That sounds reasonable indeed.